Content.Server/GameObjects/EntitySystems/Click/InteractionSystem.cs

```diff
@@ -18,6 +18,19 @@
 
 namespace Content.Server.GameObjects.EntitySystems
 {
+    /// <summary>
+    /// This interface gives components behavior when a user clicking on themselves with an object in their hand
+    /// </summary>
+    public interface IAttackSelf
```

Comment:
`attack_self` in SS13 works very differently from this, our equivalent being `IUse`.

Seems like this is poorly named. Wouldn't it make more sense to have `IAfterAttack` and such fired when attacking yourself? That's consistent with SS13 too.

Reply:
to add to this, in SS13's ```attack_self``` the "self" refers to the item.
as in the item being used on/attacked by itself.

```
/mob/proc/ClickOn( atom/A, params )
    ....
	var/obj/item/W = get_active_held_item()

	if(W == A)
		W.attack_self(src)
		update_inv_hands()
		return
```

Should definitely go with having ```IAfterAttack``` like pj says.

Reply:
killed it